src/main/java/com/google/api/codegen/metacode/FieldStructureParser.java

```diff
@@ -152,6 +177,37 @@ private static Object merge(Object mergedStructure, Object unmergedStructure) {
                 + ", this index = "
                 + index);
       }
+    } else if (unmergedStructure instanceof MapElementSpec) {
```

Comment:
I know I created a pretty long function here, but maybe it's gone too far with this addition - it might be best to separate into multiple functions.